spec/unit/provider/package/powershell_spec.rb

```diff
@@ -0,0 +1,367 @@
+#
+# Author:: Dheeraj Dubey(<dheeraj.dubey@msystechnologies.com>)
+# Copyright:: Copyright 2008-2016, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+require "chef/mixin/powershell_out"
+
+describe Chef::Provider::Package::Powershell do
+  include Chef::Mixin::PowershellOut
+  let(:timeout) { 900 }
+
+  let(:new_resource) { Chef::Resource::PowershellPackage.new("windows_test_pkg") }
+
+  let(:provider) do
+    node = Chef::Node.new
+    events = Chef::EventDispatch::Dispatcher.new
+    run_context = Chef::RunContext.new(node, {}, events)
+    Chef::Provider::Package::Powershell.new(new_resource, run_context)
+  end
+
+  let(:installed_package_stdout) do
+    <<-EOF
+"\r\nName                           Version          Source           Summary                                               \r\n----                           -------          ------           -------                                               \r\nxCertificate                   2.1.0.0          PSGallery        This module includes DSC resources that simplify administration of certificates on a Windows Server"
+    EOF
+  end
+
+  let(:package_version_stdout) do
+    <<-EOF
+"\r\nName                           Version          Source           Summary                                               \r\n----                           -------          ------           -------                                               \r\nxCertificate                   2.1.0.0          https://www.powershellgallery... PowerShellGet"
+    EOF
+  end
+
+  let(:installed_package_stdout_xnetworking) do
+    <<-EOF
+"\r\nName                           Version          Source           Summary                                               \r\n----                           -------          ------           -------                                               \r\nxNetworking                   2.12.0.0          PSGallery        Module with DSC resources for Networking Area"
+    EOF
+  end
+
+  let(:package_version_stdout_xnetworking) do
+    <<-EOF
+"\r\nName                           Version          Source           Summary                                               \r\n----                           -------          ------           -------                                               \r\nxNetworking                   2.12.0.0          https://www.powershellgallery... PowerShellGet"
+    EOF
+  end
```

Comment:
Please add a test for a package with a long name, e.g. `Security Update for Adobe Flash Player for Windows 10 Version 1607 (for x64-based Systems) (KB3194343)` so we make sure the code still works in that case.